filament/backend/test/Lifetimes.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TNT_LIFETIMES_H
+#define TNT_LIFETIMES_H
+
+#include <vector>
+#include <functional>
+#include <optional>
+
+#include "backend/Handle.h"
+#include "backend/DriverApiForward.h"
+#include "BackendTest.h"
+
+class RenderFrame {
+public:
+    explicit RenderFrame(filament::backend::DriverApi& api);
+    ~RenderFrame();
+
+    RenderFrame(RenderFrame const& rhs) noexcept = delete;
+    RenderFrame& operator=(RenderFrame const& rhs) noexcept = delete;
+    RenderFrame(RenderFrame const&& rhs) noexcept = delete;
+    RenderFrame& operator=(RenderFrame const&& rhs) noexcept = delete;
+
+private:
+    filament::backend::DriverApi& api_;
+    filament::backend::SwapChainHandle swapChain_;
+};
+
+class Cleanup {
+public:
+    explicit Cleanup(const test::BackendTest& testFixture);
+    ~Cleanup();
+
+    // These methods can't be overloaded because of a compiler error caused by the type safe casting
+    // constructor needing to be resolved for each option.
+    void AddSwapChain(filament::backend::SwapChainHandle swapChain);
```

Comment:
I had a couple optional ideas here. Maybe name all of these methods simply `add`, and overload based on the argument type. Would save some verbosity since the type of the handle will be easy to infer based on context. For example:

```
Handle<HwTexture> const srcTexture = api.createTexture(...);
cleanup.add(srcTexture);
```

You could also return the handle from the `add` method so we can do things like this:

```
Handle<HwTexture> const srcTexture = cleanup.add(api.createTexture(...));
```

Reply:
I initially wrote this as a templated `add` method but IIRC there was some weird templating issue. Due to all the random github delays it's been like 3 weeks since I wrote this so I don't remember what the issue was though, I'll change back to that and either it'll work or I'll remember what the problem was.

Reply:
Sounds good! We can also address these improvements in follow-up PRs.

Reply:
1) Haha, I had left a comment with the issue. It was that method *overloading* doesn't work because there's an implicit constructor from `Handle<A>` to `Handle<B>` if A is a base of B and that throws off the overloading.
It seems template specialization doesn't have that issue though.
2) Both were pretty easy changes to throw in.